src/components/modal/delete-modal.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import {
+  k8sDeleteResource,
+  K8sModelCommon,
+  K8sResourceCommon,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Button,
+  ButtonType,
+  ButtonVariant,
+  Form,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  TextVariants,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { Formik } from 'formik';
+import { InputField } from '../../shared';
+import { ComponentProps, createModalLauncher } from './createModalLauncher';
+
+type DeleteResourceModalProps = ComponentProps & {
+  obj: K8sResourceCommon;
+  model: K8sModelCommon;
+  description?: string;
+};
+
+export const DeleteResourceModal: React.FC<DeleteResourceModalProps> = ({
+  obj,
+  model,
+  onClose,
+  description,
+}) => {
+  const deleteResource = async () => {
+    await k8sDeleteResource({
+      model,
+      queryOptions: {
+        name: obj.metadata.name,
+        ns: obj.metadata.namespace,
+      },
+    });
+    onClose();
+  };
+
+  return (
+    <Formik onSubmit={deleteResource} initialValues={{ resourceName: '' }} onReset={onClose}>
+      {({ handleSubmit, handleReset, values, isSubmitting, setSubmitting }) => {
+        const input = values.resourceName;
+        const isValid = input === obj.metadata.name;
+        const helpText = !input ? 'missing information' : null;
+        const validatedState = !input
+          ? ValidatedOptions.warning
+          : isValid
+          ? ValidatedOptions.success
+          : ValidatedOptions.error;
+
+        return (
+          <Form>
+            <Stack hasGutter>
+              <StackItem>
+                <TextContent>
+                  {description ? (
+                    description
+                  ) : (
+                    <Text component={TextVariants.p}>
+                      The {obj.kind} <strong>{obj.metadata.name}</strong> will be deleted.
+                    </Text>
+                  )}
+                </TextContent>
+              </StackItem>
+              <StackItem>
+                <InputField
+                  name="resourceName"
+                  label={`Enter ${obj.kind} name to confirm deletion`}
+                  helpTextInvalid="Invalid input"
+                  helpText={helpText}
+                  validated={validatedState}
+                  required
+                />
+              </StackItem>
+              <StackItem>
+                <Button
+                  type={ButtonType.submit}
+                  variant={ButtonVariant.danger}
+                  isLoading={isSubmitting}
+                  onClick={(e) => {
+                    e.preventDefault();
+                    setSubmitting(true);
```

Comment:
No need to set submitting if you return a promise from handleSubmit. Formik handles it automatically.